compiler/rustc_metadata/src/creader.rs

```diff
@@ -1023,14 +1037,14 @@ impl<'a> CrateLoader<'a> {
                         dependency_of: LOCAL_CRATE,
                     },
                 );
-                cnum
+                Some(cnum)
             }
             _ => bug!(),
         }
     }
 
     pub fn process_path_extern(&mut self, name: Symbol, span: Span) -> CrateNum {
-        let cnum = self.resolve_crate(name, span, CrateDepKind::Explicit);
+        let cnum = self.resolve_crate_or_abort(name, span, CrateDepKind::Explicit);
```

Comment:
It should be very easy to return an `Option<CrateNum>` here.
The only user of this function is `fn extern_prelude_get` and it can return `self.dummy_binding` on `None`.